src/caikit_nlp_client/http_client.py

```diff
@@ -95,9 +115,22 @@ def generate_text_stream(self, model_id: str, text: str, **kwargs) -> list[str]:
         try:
             log.info(f"Calling generate_text_stream for '{model_id}'")
             json_input = create_json_request(model_id, text, **kwargs)
-            response = requests.post(  # FIXME: just a stub implementation
-                self.stream_api_url, json=json_input, timeout=10
-            )
+            if self.config.tls:
+                assert self.config.client_key_path
+                assert self.config.client_crt_path
```

Comment:
Do we need the `client_key_path` and `client_crt_path` for `tls`?  Don't we need these two for `mTLS`

Reply:
See the suggestion above